plugins/inputs/modbus/modbus.go

```diff
@@ -217,23 +217,32 @@ func (m *Modbus) InitRegister(fields []fieldContainer, name string) error {
 	sort.Slice(addrs, func(i, j int) bool { return addrs[i] < addrs[j] })
 
 	ii := 0
+	maxQuantity := 1
 	var registersRange []registerRange
+	if name == cDiscreteInputs || name == cCoils {
+		maxQuantity = 2000
+	} else if name == cInputRegisters || name == cHoldingRegisters {
+		maxQuantity = 125
+	}
 
 	// Get range of consecutive integers
 	// [1, 2, 3, 5, 6, 10, 11, 12, 14]
 	// (1, 3) , (5, 2) , (10, 3), (14 , 1)
 	for range addrs {
-		if ii < len(addrs) {
-			start := addrs[ii]
-			end := start
+		if ii >= len(addrs) {
+			break
+		}
+		quantity := 1
+		start := addrs[ii]
+		end := start
 
-			for ii < len(addrs)-1 && addrs[ii+1]-addrs[ii] == 1 {
-				end = addrs[ii+1]
-				ii++
-			}
+		for ii < len(addrs)-1 && addrs[ii+1]-addrs[ii] == 1 && quantity < maxQuantity {
+			end = addrs[ii+1]
 			ii++
-			registersRange = append(registersRange, registerRange{start, end - start + 1})
+			quantity++
 		}
+		ii++
+		registersRange = append(registersRange, registerRange{start, end - start + 1})
```

Comment:
Hmmm I'm pretty sure that the additional `i++` here means we should change `end-start+1` to `end-start`!? However, I would feel much more safe if we remove the additional `i++` and keep the range calculation as is... :-/

Reply:
we can remove `i++` but the other change _`end-start+1 `to `end-start`_  brake some test

Reply:
Hi @srebhan 

Do you have any other comments about this issue?